### ISSUE TITLE
VAAPI Failed to initialize on Ubuntu 20.04 (Or Nvidia GPU failed to show up in Emby?)

### ISSUE THREAD

**Issue by quincarter** [linuxserver/docker-emby#97]
### Is there an existing issue for this?

- [X] I have searched the existing issues
I followed #41 - VAAPI failed to initialize but nothing there helped me so I created this issue.

### Current Behavior

I'm on an amd64 based host system with Ubuntu 20.04. I have 2 SLI GTX 1080TI cards. Looks like `nvidia-smi` registers them properly on the host and the logs on the container look like the card permissions are okay. 

The actual error occurs when trying to initialize the VA API within emby itself (just looking at all the logs)

See below for all the info I have. 

## `nvidia-smi` output

![Screenshot_20240406-151256~2](https://github.com/linuxserver/docker-emby/assets/18407074/93e019fd-d3ae-4404-8d32-dcd9c651f87b)

## `/dev/dri` in the container 

![Screenshot_20240406-151350~2](https://github.com/linuxserver/docker-emby/assets/18407074/c714ac22-5dc9-435f-93a2-79828211dbda)

## Container output stating permission are okay

![Screenshot_20240406-151521~2](https://github.com/linuxserver/docker-emby/assets/18407074/bd4529a3-054e-45ab-8a33-fe32b963e41b)

## Actual error I see from the emby logs.

![Screenshot_20240406-151754~2](https://github.com/linuxserver/docker-emby/assets/18407074/37bf14d6-3923-4e5c-8c6f-ebcfeaeeb74c)

### Expected Behavior

The video cards should mount(?) and the VA API should initialize 

### Steps To Reproduce

1. Run in an Ubuntu 20.04 host machine 
2. I have 2 Nvidia GTX 1080TI with the latest Nvidia driver. (553 I think). (I tried it with a downgraded driver too (535))
3. Running the docker container using docker compose 

I have used the Nvidia-ctk to set the runtime to docker. (Not sure if I did that correctly but found quite a few references that looked like I was - I followed the steps) 

### Environment

```markdown
- OS: Ubuntu 20.04
- How docker service was installed:
docker compose
```

### CPU architecture

x86-64

### Docker creation

```yaml
emby:
    image: lscr.io/linuxserver/emby:latest
    container_name: emby
   #  network_mode: host
    environment:
      - PUID=${EMBY_UID}
      - PGID=${EMBY_GID}
      - NVIDIA_VISIBLE_DEVICES=all
      - NVIDIA_DRIVER_CAPABILITIES=all
      - TZ=${TIMEZONE}
    volumes:
      - ./.containers/Emby:/config # Configuration directory
      - ./images/logowhite.png:/app/emby/system/dashboard-ui/modules/logoscreensaver/logowhite.png ## Uncomment and create your own logo in the image directory to customize your server
      - ./images/logowhite.png:/app/emby/system/dashboard-ui/modules/themes/logowhite.png ## Uncomment and create your own logo in the image directory to customize your server
      - ./images/logodark.png:/app/emby/system/dashboard-ui/modules/themes/logodark.png ## Uncomment and create your own logo in the image directory to customize your server
      - ${LOCAL_TV_PATH}:/media/Synology/Television # Media directory
      - ${LOCAL_MOVIES_PATH}:/media/Synology/Movies # Media directory
      - ${LOCAL_BACKUPS_PATH}:/media/Synology/Backups #Backups Directory
     #  - /ssl:/ssl # Add SSL Certs to this directory
      - /ssl/fullchain.pem:/ssl/fullchain.pem
      - /ssl/privkey.pem:/ssl/privkey.pem
      - /ssl/token:/ssl/token
    ports:
      - ${EMBY_HOST_PORT}:8096 #http port
      - ${EMBY_HOST_PORT_SSL}:8920 #ssl port
    runtime: nvidia # Expose NVIDIA GPUs if any (Apple Silicon does not use this)
    # network_mode: host # Enable DLNA and Wake-on-Lan
    devices:
      - /dev/dri:/dev/dri # VAAPI/NVDEC/NVENC render nodes
    #   - /dev/vchiq:/dev/vchiq # MMAL/OMX on Raspberry Pi
    #   - /dev/video10:/dev/video10 #optional other video cards
    #   - /dev/video11:/dev/video11 #optional
    #   - /dev/video12:/dev/video12 #optional
    restart: unless-stopped
    profiles:
      - emby
```

### Container logs

```bash
➜  ~ docker logs emby                                                                     [migrations] started
[migrations] no migrations found
───────────────────────────────────────

      ██╗     ███████╗██╗ ██████╗
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝

   Brought to you by linuxserver.io
───────────────────────────────────────

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    1000
User GID:    1000
───────────────────────────────────────

**** adding /dev/dri/renderD128 to video group root with id 0 ****
**** permissions for /dev/dri/card1 are good ****
**** permissions for /dev/dri/card0 are good ****
**** permissions for /dev/dri/renderD129 are good ****
[custom-init] No custom files found, skipping...
Info Main: Application path: /app/emby/system/EmbyServer.dll
Info App: Setting default culture to en-US
Info Main: Emby
        Command line: /app/emby/system/EmbyServer.dll -programdata /config -ffdetect /app/emby/bin/ffdetect -ffmpeg /app/emby/bin/ffmpeg -ffprobe /app/emby/bin/ffprobe -restartexitcode 3
        Operating system: Linux version 5.15.0-101-generic (buildd@lcy02-amd64-032) (gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #111-Ubun
        Framework: .NET 6.0.25
        OS/Process: x64/x64
        Runtime: app/emby/system/System.Private.CoreLib.dll
        Processor count: 16
        Data path: /config
        Application path: /app/emby/system
Info Main: Logs path: /config/logs
Info Main: Cache path: /config/cache
Info Main: Internal metadata path: /config/metadata
Info App: Emby Server Version: 4.8.3.0
Info App: Loading assemblies
Info App: File /config/plugins/Emby.M3UTuner.dll has version 1.0.29.0
Info App: File /app/emby/system/plugins/Emby.M3UTuner.dll has version 1.0.28.0
Info App: File /config/plugins/Tvdb.dll has version 1.4.6.0
Info App: File /app/emby/system/plugins/Tvdb.dll has version 1.4.5.0
Info App: File /config/plugins/Emby.Server.CinemaMode.dll has version 1.0.47.0
Info App: File /app/emby/system/plugins/Emby.Server.CinemaMode.dll has version 1.0.47.0
Info App: File /config/plugins/Emby.PortMapper.dll has version 1.2.4.0
Info App: File /app/emby/system/plugins/Emby.PortMapper.dll has version 1.2.4.0
Info App: File /config/plugins/OMDb.dll has version 1.0.21.0
Info App: File /app/emby/system/plugins/OMDb.dll has version 1.0.21.0
Info App: File /config/plugins/StudioImages.dll has version 1.0.3.0
Info App: File /app/emby/system/plugins/StudioImages.dll has version 1.0.3.0
Info App: File /config/plugins/MusicBrainz.dll has version 1.0.22.0
Info App: File /app/emby/system/plugins/MusicBrainz.dll has version 1.0.22.0
Info App: File /config/plugins/Emby.Dlna.dll has version 1.3.8.0
Info App: File /app/emby/system/plugins/Emby.Dlna.dll has version 1.3.7.0
Info App: File /config/plugins/NfoMetadata.dll has version 1.0.80.0
Info App: File /app/emby/system/plugins/NfoMetadata.dll has version 1.0.80.0
Info App: File /config/plugins/Fanart.dll has version 1.0.16.0
Info App: File /app/emby/system/plugins/Fanart.dll has version 1.0.16.0
Info App: File /config/plugins/BlurayMounter.dll has version 1.0.2.0
Info App: File /app/emby/system/plugins/BlurayMounter.dll has version 1.0.2.0
Info App: File /config/plugins/DvdMounter.dll has version 1.0.0.0
Info App: File /app/emby/system/plugins/DvdMounter.dll has version 1.0.0.0
Info App: File /config/plugins/MovieDb.dll has version 1.7.4.0
Info App: File /app/emby/system/plugins/MovieDb.dll has version 1.7.4.0
Info App: File /config/plugins/MBBackup.dll has version 1.6.3.0
Info App: File /app/emby/system/plugins/MBBackup.dll has version 1.6.3.0
Info App: File /config/plugins/OpenSubtitles.dll has version 1.0.54.0
Info App: File /app/emby/system/plugins/OpenSubtitles.dll has version 1.0.53.0
Info App: File /config/plugins/AudioDb.dll has version 1.0.18.0
Info App: File /app/emby/system/plugins/AudioDb.dll has version 1.0.18.0
Info App: File /config/plugins/Emby.Webhooks.dll has version 1.0.35.0
Info App: File /app/emby/system/plugins/Emby.Webhooks.dll has version 1.0.35.0
Info App: File /config/plugins/Emby.XmlTV.dll has version 1.1.8.0
Info App: File /app/emby/system/plugins/Emby.XmlTV.dll has version 1.1.8.0
Info App: File /config/plugins/EmbyGuideData.dll has version 1.0.13.0
Info App: File /app/emby/system/plugins/EmbyGuideData.dll has version 1.0.13.0
Info App: Loading Emby.M3UTuner, Version=1.0.29.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Emby.M3UTuner.dll
Info App: Loading Tvdb, Version=1.4.6.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Tvdb.dll
Info App: Loading Emby.Server.CinemaMode, Version=1.0.47.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Emby.Server.CinemaMode.dll
Info App: Loading Trakt, Version=3.7.2.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Trakt.dll
Info App: Loading Emby.PortMapper, Version=1.2.4.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Emby.PortMapper.dll
Info App: Loading OMDb, Version=1.0.21.0, Culture=neutral, PublicKeyToken=null from /config/plugins/OMDb.dll
Info App: Loading MediaBrowser.Plugins.Trailers, Version=1.3.8.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Mediabrowser.Plugins.Trailers.dll
Info App: Loading StudioImages, Version=1.0.3.0, Culture=neutral, PublicKeyToken=null from /config/plugins/StudioImages.dll
Info App: Loading MusicBrainz, Version=1.0.22.0, Culture=neutral, PublicKeyToken=null from /config/plugins/MusicBrainz.dll
Info App: Loading Emby.Dlna, Version=1.3.8.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Emby.Dlna.dll
Info App: Loading NfoMetadata, Version=1.0.80.0, Culture=neutral, PublicKeyToken=null from /config/plugins/NfoMetadata.dll
Info App: Loading Fanart, Version=1.0.16.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Fanart.dll
Info App: Loading BlurayMounter, Version=1.0.2.0, Culture=neutral, PublicKeyToken=null from /config/plugins/BlurayMounter.dll
Info App: Loading DvdMounter, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null from /config/plugins/DvdMounter.dll
Info App: Loading MovieDb, Version=1.7.4.0, Culture=neutral, PublicKeyToken=null from /config/plugins/MovieDb.dll
Info App: Loading MBBackup, Version=1.6.3.0, Culture=neutral, PublicKeyToken=null from /config/plugins/MBBackup.dll
Info App: Loading OpenSubtitles, Version=1.0.54.0, Culture=neutral, PublicKeyToken=null from /config/plugins/OpenSubtitles.dll
Info App: Loading AudioDb, Version=1.0.18.0, Culture=neutral, PublicKeyToken=null from /config/plugins/AudioDb.dll
Info App: Loading Emby.Webhooks, Version=1.0.35.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Emby.Webhooks.dll
Info App: Loading Emby.XmlTV, Version=1.1.8.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Emby.XmlTV.dll
Info App: Loading EmbyGuideData, Version=1.0.13.0, Culture=neutral, PublicKeyToken=null from /config/plugins/EmbyGuideData.dll
Info App: Loading Emby.Api, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Web, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading MediaBrowser.Model, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading MediaBrowser.Common, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading MediaBrowser.Controller, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Providers, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Photos, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Server.Implementations, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.LiveTV, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.ActivityLog, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Server.MediaEncoding, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.LocalMetadata, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Notifications, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Web.GenericUI, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Codecs.Dxva, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Codecs, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Server.Connect, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading Emby.Server.Sync, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info App: Loading EmbyServer, Version=4.8.3.0, Culture=neutral, PublicKeyToken=null
Info SqliteUserRepository: Sqlite version: 3.42.0
Info SqliteUserRepository: Sqlite compiler options: ATOMIC_INTRINSICS=1,COMPILER=gcc-10.3.0,DEFAULT_AUTOVACUUM,DEFAULT_CACHE_SIZE=-2000,DEFAULT_FILE_FORMAT=4,DEFAULT_JOURNAL_SIZE_LIMIT=-1,DEFAULT_MMAP_SIZE=0,DEFAULT_PAGE_SIZE=4096,DEFAULT_PCACHE_INITSZ=20,DEFAULT_RECURSIVE_TRIGGERS,DEFAULT_SECTOR_SIZE=4096,DEFAULT_SYNCHRONOUS=2,DEFAULT_WAL_AUTOCHECKPOINT=1000,DEFAULT_WAL_SYNCHRONOUS=2,DEFAULT_WORKER_THREADS=0,ENABLE_COLUMN_METADATA,ENABLE_DBSTAT_VTAB,ENABLE_FTS3,ENABLE_FTS3_PARENTHESIS,ENABLE_FTS3_TOKENIZER,ENABLE_FTS4,ENABLE_FTS5,ENABLE_GEOPOLY,ENABLE_MATH_FUNCTIONS,ENABLE_PREUPDATE_HOOK,ENABLE_RTREE,ENABLE_SESSION,ENABLE_UNLOCK_NOTIFY,ENABLE_UPDATE_DELETE_LIMIT,LIKE_DOESNT_MATCH_BLOBS,MALLOC_SOFT_LIMIT=1024,MAX_ATTACHED=10,MAX_COLUMN=2000,MAX_COMPOUND_SELECT=500,MAX_DEFAULT_PAGE_SIZE=8192,MAX_EXPR_DEPTH=1000,MAX_FUNCTION_ARG=127,MAX_LENGTH=1000000000,MAX_LIKE_PATTERN_LENGTH=50000,MAX_MMAP_SIZE=0x7fff0000,MAX_PAGE_COUNT=1073741823,MAX_PAGE_SIZE=65536,MAX_SCHEMA_RETRY=25,MAX_SQL_LENGTH=1000000000,MAX_TRIGGER_DEPTH=1000,MAX_VARIABLE_NUMBER=250000,MAX_VDBE_OP=250000000,MAX_WORKER_THREADS=8,MUTEX_PTHREADS,OMIT_LOOKASIDE,SECURE_DELETE,SYSTEM_MALLOC,TEMP_STORE=1,THREADSAFE=1
Info SqliteUserRepository: Opening sqlite connection to /config/data/users.db
Info SqliteUserRepository: Default journal_mode for /config/data/users.db is wal
Info SqliteUserRepository: PRAGMA foreign_keys=1
Info SqliteUserRepository: Result of setting SQLITE_DBCONFIG_DQS_DDL to 0 is 0
Info SqliteUserRepository: Result of setting SQLITE_DBCONFIG_DQS_DML to 0 is 0
Info ActivityRepository: Opening sqlite connection to /config/data/activitylog.db
Info ActivityRepository: Default journal_mode for /config/data/activitylog.db is wal
Info ActivityRepository: PRAGMA foreign_keys=1
Info ActivityRepository: Result of setting SQLITE_DBCONFIG_DQS_DDL to 0 is 0
Info ActivityRepository: Result of setting SQLITE_DBCONFIG_DQS_DML to 0 is 0
Info NetworkManager: Detecting local network addresses
Info NetworkManager: networkInterface: Ethernet eth0, Speed: 10000000000, Description: eth0
Info NetworkManager: GatewayAddresses: 172.21.0.1
Info NetworkManager: UnicastAddresses: 172.21.0.4,fe80::42:acff:fe15:4%41
Info NetworkManager: networkInterface: Loopback lo, Speed: -1, Description: lo
Info NetworkManager: GatewayAddresses:
Info NetworkManager: UnicastAddresses: 127.0.0.1,::1
Info NetworkManager: Detected local ip addresses: [{"IPAddress":"172.21.0.4","HasGateWayAddress":true,"PrefixLength":16,"IPv4Mask":"255.255.0.0"},{"IPAddress":"fe80::42:acff:fe15:4%41","HasGateWayAddress":true,"PrefixLength":64},{"IPAddress":"127.0.0.1","HasGateWayAddress":false,"PrefixLength":8,"IPv4Mask":"255.0.0.0"},{"IPAddress":"::1","HasGateWayAddress":false,"PrefixLength":128}]
Info SqliteDisplayPreferencesRepository: Opening sqlite connection to /config/data/displaypreferences.db
Info SqliteDisplayPreferencesRepository: Default journal_mode for /config/data/displaypreferences.db is wal
Info SqliteDisplayPreferencesRepository: PRAGMA foreign_keys=1
Info SqliteDisplayPreferencesRepository: Result of setting SQLITE_DBCONFIG_DQS_DDL to 0 is 0
Info SqliteDisplayPreferencesRepository: Result of setting SQLITE_DBCONFIG_DQS_DML to 0 is 0
Info ServerConfigurationManager: Saving system configuration
Info App: Begin vacumming SqliteItemRepository
Info SqliteItemRepository: Opening sqlite connection to /config/data/library.db
Info SqliteItemRepository: Default journal_mode for /config/data/library.db is wal
Info SqliteItemRepository: PRAGMA cache_size=-3072000
Info SqliteItemRepository: PRAGMA page_size=4096
Info SqliteItemRepository: PRAGMA foreign_keys=1
Info SqliteItemRepository: Result of setting SQLITE_DBCONFIG_DQS_DDL to 0 is 0
Info SqliteItemRepository: Result of setting SQLITE_DBCONFIG_DQS_DML to 0 is 0
Info App: Completed vacumming SqliteItemRepository after 5725ms
Info App: Begin vacumming SqliteUserRepository
Info App: Completed vacumming SqliteUserRepository after 648ms
Info App: Begin vacumming AuthenticationRepository
Info AuthenticationRepository: Opening sqlite connection to /config/data/authentication.dbInfo AuthenticationRepository: Default journal_mode for /config/data/authentication.db is wal
Info AuthenticationRepository: PRAGMA foreign_keys=1
Info AuthenticationRepository: Result of setting SQLITE_DBCONFIG_DQS_DDL to 0 is 0
Info AuthenticationRepository: Result of setting SQLITE_DBCONFIG_DQS_DML to 0 is 0
Info App: Completed vacumming AuthenticationRepository after 836ms
Info App: Begin vacumming SqliteDisplayPreferencesRepository
Info App: Completed vacumming SqliteDisplayPreferencesRepository after 1110ms
Info App: Begin vacumming ActivityRepository
Info App: Completed vacumming ActivityRepository after 691ms
Info App: Adding HttpListener prefix http://+:8096/
Info App: Adding HttpListener prefix https://+:8920/
Info SqliteItemRepository: Init Complete
Info App: Emby
        Command line: /app/emby/system/EmbyServer.dll -programdata /config -ffdetect /app/emby/bin/ffdetect -ffmpeg /app/emby/bin/ffmpeg -ffprobe /app/emby/bin/ffprobe -restartexitcode 3
        Operating system: Linux version 5.15.0-101-generic (buildd@lcy02-amd64-032) (gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #111-Ubun
        Framework: .NET 6.0.25
        OS/Process: x64/x64
        Runtime: app/emby/system/System.Private.CoreLib.dll
        Processor count: 16
        Data path: /config
        Application path: /app/emby/system
Info App: Logs path: /config/logs
Info App: Cache path: /config/cache
Info App: Internal metadata path: /config/metadata
Info App: Transcoding temporary files path: /config/transcoding-temp
[ls.io-init] done.
Info FfmpegManager: FFMpeg: /app/emby/bin/ffmpeg
Info FfmpegManager: FFProbe: /app/emby/bin/ffprobe
Info FfmpegManager: FFDetect: /app/emby/bin/ffdetect
Info Skia: SkiaSharp version: 2.88.0.0
Info ImageProcessor: Adding image processor Skia
Info libvips: NetVips version: 2.3.1.0
Info ImageProcessor: Adding image processor libvips
Info TaskManager: Daily trigger for Emby Server Backup set to fire at 04/07/2024 00:10:00, which is 1438.948595525 minutes from now.
Info TaskManager: Daily trigger for Thumbnail image extraction set to fire at 04/06/2024 02:00:00, which is 108.94839632333333 minutes from now.
Info TaskManager: Daily trigger for Rotate log file set to fire at 04/07/2024 00:00:00, which is 1428.9453180716666 minutes from now.
Info TaskManager: Queueing task HardwareDetectionScheduledTask
Info TaskManager: Executing Hardware Detection
Info App: ServerId: 9788d2f967c5430689e6cb69e8ed80fd
Info FfmpegManager: ProcessRun 'ffmpeg -hide_banner -version' Execute: /app/emby/bin/ffmpeg -hide_banner -version
Info App: Starting entry point Emby.Dlna.Main.DlnaEntryPoint
Info FfmpegManager: ProcessRun 'ffmpeg -hide_banner -version' Process exited with code 0 - Succeeded
Info FfmpegManager: FfmpegValidator.Validate complete
Inf

```

**Comment by j0nnymoe** [comment]
Any reason you're using vaapi Vs nvenc which is supported in emby?

**Comment by aptalca** [comment]
https://github.com/linuxserver/docker-emby?tab=readme-ov-file#nvidia

Readme doesn't say to map /dev/dri for Nvidia

**Comment by quincarter** [comment]
> https://github.com/linuxserver/docker-emby?tab=readme-ov-file#nvidia
> 
> Readme doesn't say to map /dev/dri for Nvidia

I guess to be fair it doesn't say what exactly to map for my device(s) (exactly how to do it)

I have had this configured for a while and it was working in emby. Can you share your configuration for Nvidia cards?

**Comment by Roxedus** [comment]
You dont need a explicit mapping for nvidia cards, the toolkit/runtime handles that for you

**Comment by quincarter** [comment]
Or should I just comment out the cards mapping?

**Comment by quincarter** [comment]
![Screenshot_20240406-155541.png](https://github.com/linuxserver/docker-emby/assets/18407074/3ded48a9-5004-4741-a1ca-1b5e99ca4743)

So why aren't any encoding options showing up here?

**Comment by j0nnymoe** [comment]
Do you get `nvidia-smi` from within the container?

**Comment by aptalca** [comment]
To be clear, we don't do anything for Nvidia other than setting one env var (capabilities) in the image. Everything is handled by Nvidia toolkit and the drivers installed on host. 

If they're installed correctly, the container is using Nvidia runtime and the other env is set, it should work. But again, none of that is under our control. 

Emby displays all valid gpu options it detects in its gui.

**Comment by quincarter** [comment]
> Do you get `nvidia-smi` from within the container?

actually, no i don't

```bash
docker exec -it emby /bin/bash
root@de7ec57a98a0:/# nvidia-smi
Failed to initialize NVML: Unknown Error
```

**Comment by quincarter** [comment]
So why would i not have `nvidia-smi` in the container?

**Comment by j0nnymoe** [comment]
Have you installed the nvidia container toolkit?

**Comment by quincarter** [comment]
> Have you installed the nvidia container toolkit?

Yeah multiple times i feel like. 

```bash
nvidia-ctk -v
NVIDIA Container Toolkit CLI version 1.14.6
commit: 5605d191332dcfeea802c4497360d60a65c7887e
```

also my current `/etc/docker/daemon.json` and my `~/.config/docker/daemon.json` are identical:

```json
{
  "default-runtime": "nvidia",
  "runtimes": {
    "nvidia": {
      "args": [],
      "path": "nvidia-container-runtime"
      }
    },
    "exec-opts": ["native.cgroupdriver=cgroupfs"]
}
```

**Comment by quincarter** [comment]
this was the command i ran to configure the `nvidia-ctk` runtime:

```bash
nvidia-ctk runtime configure --runtime=docker --config=$HOME/.config/docker/daemon.json
```

**Comment by j0nnymoe** [comment]
How did you install docker? 

Note - it might be better you to swing over to our discord for easier help. Github issues are mainly for bugs where this seems to be an issue with your host.

**Comment by quincarter** [comment]
> How did you install docker?
> 
> Note - it might be better you to swing over to our discord for easier help. Github issues are mainly for bugs where this seems to be an issue with your host.

Sure. Which channel should i post in? I just joined

**Comment by j0nnymoe** [comment]
#container-support

**Comment by quincarter** [comment]
Linking here for visibility in case someone finds this searching - added a discord post here:
https://discord.com/channels/354974912613449730/1226289649530568725

**Comment by quincarter** [comment]
I do see in the most recent diff comparing the two versions [current and previous](https://github.com/linuxserver/docker-emby/compare/4.8.3.0-ls196...beta-4.8.0.19-ls197) -- that the `NVIDIA_DRIVER_CAPABILITIES` is now being set by the docker image instead of being specified in the docker compose file. [See this](https://github.com/linuxserver/docker-emby/blob/10a56a4cb25fcf73c2ff7fa726662522136a76a2/Dockerfile#L47) you can see this is changed. I currently set this to `NVIDIA_DRIVER_CAPABILITIES=all` in my docker compose.  I know that in order to use `nvidia-smi` in the docker container, you need the `utility` option.  [See Driver Capabilities docs](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/1.10.0/user-guide.html#driver-capabilities)

**Comment by quincarter** [comment]
TLDR - solved this in Discord for anyone that comes across this in google and wants to try this solution. 

The container wouldn't initialize the `nvidia-smi` and any cards associated with it. So mapping the nvidia stuff directly (as seen in the command above in the docker command) is what ultimately solved it. 
>--device=/dev/nvidia-uvm \
    --device=/dev/nvidia-uvm-tools \
    --device=/dev/nvidia-modeset \
    --device=/dev/nvidiactl \
    --device=/dev/nvidia0 \
    --device=/dev/nvidia1 \

In my docker compose it looks like this (thanks @salty):

```yaml
  emby:
    image: lscr.io/linuxserver/emby:latest
    container_name: emby
    environment:
      - PUID=${EMBY_UID}
      - PGID=${EMBY_GID}
      - NVIDIA_VISIBLE_DEVICES=all
      - NVIDIA_DRIVER_CAPABILITIES=all
      - TZ=${TIMEZONE}
    volumes:
      - ./.containers/Emby:/config # Configuration directory
      - ./images/logowhite.png:/app/emby/system/dashboard-ui/modules/logoscreensaver/logowhite.png 
      - ./images/logowhite.png:/app/emby/system/dashboard-ui/modules/themes/logowhite.png 
      - ./images/logodark.png:/app/emby/system/dashboard-ui/modules/themes/logodark.png 
      - ${LOCAL_TV_PATH}:/media/Synology/Television # Media directory
      - ${LOCAL_MOVIES_PATH}:/media/Synology/Movies # Media directory
      - ${LOCAL_BACKUPS_PATH}:/media/Synology/Backups #Backups Directory
      - /ssl/fullchain.pem:/ssl/fullchain.pem
      - /ssl/privkey.pem:/ssl/privkey.pem
      - /ssl/token:/ssl/token
    ports:
      - ${EMBY_HOST_PORT}:8096 #http port
      - ${EMBY_HOST_PORT_SSL}:8920 #ssl port
    runtime: nvidia
    restart: unless-stopped
    devices:
      - "/dev/nvidia-uvm:/dev/nvidia-uvm" # Added nvidia devices here
      - "/dev/nvidia-uvm-tools:/dev/nvidia-uvm-tools" # Added nvidia devices here
      - "/dev/nvidia-modeset:/dev/nvidia-modeset" # Added nvidia devices here
      - "/dev/nvidiactl:/dev/nvidiactl" # Added nvidia devices here
      - "/dev/nvidia0:/dev/nvidia0" # Added nvidia devices here
      - "/dev/nvidia1:/dev/nvidia1" # Added nvidia devices here (i have a 2nd GPU so i needed this too)
      - "/dev/dri:/dev/dri" # I added this per suggestion, but this is for VAAPI so i don't know if this actually works -- this was what was failing before. 
    profiles:
      - emby
```

**Comment by quincarter** [comment]
I think this is solved per the suggestion above. So i am closing this now! Thanks for everyone's help! I am not sure why i had to add the devices manually in the docker compose but that's what i had to do to get them to show up now.
